### DISCUSSION TITLE
How to convert date column provided as string type of `DD.MM.YYYY` to standard `YYYY-MM-DD`?

### DISCUSSION THREAD

**Discussion by emmansh** [duckdb/duckdb#5363] [Q&A] [answered]
I have a dataset with a date column provided as a string in the format of `DD.MM.YYYY`. I want to modify this column to type `DATE` in the standard `YYYY-MM-DD`. I have tried following [this answer](https://stackoverflow.com/a/34282815/6105259) that relies on Postgres's [`to_date()`](https://www.postgresql.org/docs/current/functions-formatting.html), but it doesn't work.

Example
---

```sql
-- duckdb
CREATE TABLE my_table (
  name varchar(20) NULL,
  lastname varchar(20) NULL,
  dob varchar(20) NULL
);

INSERT INTO my_table (name, lastname, dob)
  VALUES ('john', 'smith', '13.11.1994'),
  ('mike', 'jones', '26.12.1999');
```

The immidiate first step would be to replace `.` with `-`, so we get `dd-mm-yyyy`:
```sql
-- duckdb
SELECT
  REPLACE(dob, '.', '-') AS date_with_dashes
FROM my_table;

/*
┌──────────────────┐
│ date_with_dashes │
╞══════════════════╡
│ 13-11-1994       │
│ 26-12-1999       │
└──────────────────┘
*/
```
But now what? I have a column of type `VARCHAR` in the format `DD-MM-YYYY`. How do I reverse it to `YYYY-MM-DD` _and_ make it of type `DATE`?

[This answer](https://stackoverflow.com/a/34282815/6105259) suggests using `to_date()`, which is a postgres function that unfortunately duckdb doesn't accept.
```sql
-- duckdb
SELECT
  to_date(REPLACE(dob, '.', '-'), 'dd-mm-yyyy') AS date_fixed
FROM my_table;
```

> ```
> Catalog Error: Scalar Function with name to_date does not exist!
> Did you mean "to_days"?
> LINE 1: select to_date(replace(dob, '.', '-'), 'dd-mm-...
> ```

<br>

What other processing options duckdb would support given my dataset provided with `DD.MM.YYYY`?

**Comment by Mytherin** [comment, marked as answer]
Try [strptime](https://duckdb.org/docs/sql/functions/dateformat#strptime-examples), e.g.:

```sql
select strptime('20-10-1992', '%d-%m-%Y');
```
